satpy/readers/ahi_hsd.py

```diff
@@ -419,12 +419,12 @@ def end_time(self):
     @property
     def observation_start_time(self):
         """Get the observation start time."""
-        return datetime(1858, 11, 17) + timedelta(days=float(self.basic_info["observation_start_time"]))
+        return datetime(1858, 11, 17) + timedelta(days=float(self.basic_info["observation_start_time"][0]))
```

Comment:
I think it would be preferable to use `.item()` if possible. That method will raise an error if the underlying array is not of size 1.

Reply:
Done!

Reply:
Oh, I meant to use item everywhere... Or does it not work in the other places?

Reply:
It *should* work in the other places..but in some edge cases many of the other places could (in theory) be a multidimensional array so I was reluctant to drop `item` in on those.

Reply:
ok.

Reply:
Although, in any case, it is best for us to make sure there is only one value right? Because we only support that case. I would prefer the reader to fail when that happens, as we don't explicitly handle the case.

Reply:
Yeah, good point. I changed them all over.